pytorch_lightning/core/datamodule.py

```diff
@@ -262,3 +267,72 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
             state_dict: the datamodule state returned by ``state_dict``.
         """
         pass
+
+    @classmethod
+    def load_from_checkpoint(
+        cls,
+        checkpoint_path: Union[_PATH, IO],
+        hparams_file: Optional[_PATH] = None,
+        **kwargs,
+    ):
+        r"""
+        Primary way of loading a datamodule from a checkpoint. When Lightning saves a checkpoint
+        it stores the arguments passed to ``__init__``  in the checkpoint under ``"datamodule_hyper_parameters"``.
+
+        Any arguments specified through \*\*kwargs will override args stored in ``"datamodule_hyper_parameters"``.
+
+        Args:
+            checkpoint_path: Path to checkpoint. This can also be a URL, or file-like object
+            hparams_file: Optional path to a ``.yaml`` or ``.csv`` file with hierarchical structure
+                as in this example::
+
+                    dataloader:
+                        batch_size: 32
+
+                You most likely won't need this since Lightning will always save the hyperparameters
+                to the checkpoint.
+                However, if your checkpoint weights don't have the hyperparameters saved,
+                use this method to pass in a ``.yaml`` file with the hparams you'd like to use.
+                These will be converted into a :class:`~dict` and passed into your
+                :class:`LightningDataModule` for use.
+
+                If your datamodule's ``hparams`` argument is :class:`~argparse.Namespace`
+                and ``.yaml`` file has hierarchical structure, you need to refactor your datamodule to treat
+                ``hparams`` as :class:`~dict`.
+            \**kwargs: Any extra keyword args needed to init the datamodule. Can also be used to override saved
```

Comment:
I don't think this is required? At least we don't use it everywhere else

```suggestion
            **kwargs: Any extra keyword args needed to init the datamodule. Can also be used to override saved
```

Reply:
in wandb we do: https://pytorch-lightning.readthedocs.io/en/latest/extensions/generated/pytorch_lightning.loggers.WandbLogger.html#wandblogger and it correctly reflects what `kwargs` are here for.

else it displays it like a normal argument.
https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#pytorch_lightning.core.lightning.LightningModule.load_from_checkpoint.params.kwargs